falcon_multiqc/commands/save.py

```diff
@@ -21,156 +22,182 @@
+    # TODO call this function for a directory and sample_name
+    directory_name = basename(directory)
+    sample_metadata_name = basename(sample_metadata)
+    click.echo(f'Saving: {directory_name} with sample metadata: {sample_metadata_name}...')
+                
+    with open(directory + "/multiqc_data/multiqc_data.json") as multiqc_data:
+
+        with open(sample_metadata) as sample_metadata:
+            # Skip header
+            next(sample_metadata)
+
+            # Keep track of samples added, so we know its primary key, when saving raw data later.
+            samples = {}  # name : primary key id
+            batches = [] # batches within given metadata 
+
+            # Cohort id for this input. Cohort id must be the same for every batch of this input.
+            cohort_id = None
+
+            for line in sample_metadata:
+                split = line.split(",")
+                try:
+                    sample_name = split[0]
+                    batch_name = split[2]
+                    flowcell_lane = split[3]
+                    library_id = split[4]
+                    platform = split[5]
+                    centre = split[6]
+                    reference = split[7]
+                    type = split[8]
+                    description = split[9]
+                except IndexError:
+                    raise Exception(f"Metadata format is invalid, Accepted format is:"
+                    "\n'Sample Name' 'Cohort Name' 'Batch Name' 'Flowcell.Lane' 'Library ID' 'Platform' 'Centre of Sequencing' 'Reference Genome' 'type' 'Description'")
+
+                if not cohort_id:
+                    # Get cohort id / name from the first data row (assuming the metadata is for 1 cohort).
+                    cohort_id = split[1]
+                    if session.query(Cohort.id).filter_by(id=cohort_id).scalar() is None:
+                        # Cohort does not exist in database.
+                        cohort_row = Cohort(
+                            id=cohort_id,
+                            description=cohort_description
+                        )
+                        session.add(cohort_row)
+                elif split[1] != cohort_id:
+                    raise Exception(f"Metadata input has multiple cohort ids ({cohort_id} and {split[1]}). Save supports one cohort at a time.")
+                
+                batch_id = None
+                if batch_name not in batches:
+                    # First time this batch_name is seen from this given metadata.csv
+                    batches.append(batch_name) 
+                    if session.query(Batch.id).filter(Batch.batch_name == batch_name, Batch.cohort_id == cohort_id).scalar() is None:
+                        # Batch does not exist in database.
+                        batch_row = Batch(
+                            cohort_id=cohort_id,
+                            batch_name=batch_name,
+                            path=abspath(directory),
```

Comment:
What do you think about moving this outside this function? And instead having it called when passing it in as the argument, so directory in this method we know is an absolute path.

Or up the top of this function. 

Basically I'm just trying to double check that all the paths we deal with are absolute to avoid issues.

Reply:
so am i right to put it at the top of the function, since each time the function is called the directory will be different

Reply:
do you mean change all the times i reference directory to be using `path` which is `abs(directory)`?

Reply:
Yeah just put it up the top, directory = abs(directory) if you want. Or I just meant instead when calling this function save_sample(x, y, abs(directory))

Reply:
oh yep that's even better - shall do :)